HeterogeneousCore/SonicTriton/interface/TritonClient.h

```diff
@@ -0,0 +1,93 @@
+#ifndef HeterogeneousCore_SonicTriton_TritonClient
+#define HeterogeneousCore_SonicTriton_TritonClient
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "HeterogeneousCore/SonicCore/interface/SonicClientSync.h"
+#include "HeterogeneousCore/SonicCore/interface/SonicClientPseudoAsync.h"
+#include "HeterogeneousCore/SonicCore/interface/SonicClientAsync.h"
+
+#include <vector>
+#include <string>
+#include <exception>
+
+#include "request_grpc.h"
+
+namespace ni = nvidia::inferenceserver;
+namespace nic = ni::client;
+
+template <typename Client>
+class TritonClient : public Client {
+public:
+  struct ServerSideStats {
+    uint64_t request_count_;
+    uint64_t cumul_time_ns_;
+    uint64_t queue_time_ns_;
+    uint64_t compute_time_ns_;
+  };
+
+  //constructor
+  TritonClient(const edm::ParameterSet& params);
+
+  //helper
+  bool getResults(const std::unique_ptr<nic::InferContext::Result>& result);

```

Comment:
Why not just
```suggestion
  bool getResults(const nic::InferContext::Result& result);
```
since ownership is not transferred?

Reply:
Requires a change elsewhere as well (because of the Triton API), but will do